eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -155,43 +155,45 @@ jobs:
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'spotbugs:check checkstyle:checkstyle-aggregate'
+          goals: 'spotbugs:check checkstyle:checkstyle-aggregate javadoc:aggregate'
 
-      # Generate the Maven site report, including SpotBugs, JavaDoc, CheckStyle, Jacoco, etc only when we are not
-      # running as a PR check.
+      # We run a separate SpotBugs aggregate report step here to roll-up all the issues identified per-module
       - task: Maven@3
-        displayName: 'Generate Maven project site, including JavaDocs, SpotBugs, and CheckStyle reports'
+        displayName: 'Run SpotBugs aggregate analysis'
         inputs:
-          mavenPomFile: pom.client.xml
-          options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
+          mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
+          options: '--batch-mode -DskipTests -Dgpg.skip'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'install site:site site:stage'
+          goals: 'install'
 
-      # We run a separate SpotBugs aggregate report step here to roll-up all the issues identified per-module in the
-      # previous step.
+      # Generate the Maven site report, including SpotBugs, JavaDoc, CheckStyle, Jacoco, etc only when we are not
+      # running as a PR check.
       - task: Maven@3
-        displayName: 'Run SpotBugs'
+        displayName: 'Generate Maven project site, including JavaDocs, SpotBugs, and CheckStyle reports'
+        condition: ne(variables['Build.Reason'], 'PullRequest')
         inputs:
-          mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
-          options: '--batch-mode -DskipTests -Dgpg.skip'
+          mavenPomFile: pom.client.xml
+          options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'install'
+          goals: 'install site:site site:stage'
 
       - script: |
           git clone https://github.com/JonathanGiles/DependencyChecker.git
           mkdir input && cp dependencies.json input/
         displayName: 'Download dependency checker'
+        condition: ne(variables['Build.Reason'], 'PullRequest')
 
       - task: Maven@3
         displayName: 'Analyze dependencies'
+        condition: ne(variables['Build.Reason'], 'PullRequest')
```

Comment:
We need to be a little bit careful about shifting things that should be detected out of the PR process.

Reply:
At this point in time the analyse dependencies step only creates a report. When it can detect violations and fail the build, we should definitely enable this to run all the time.

Reply:
Looks like we have issue https://github.com/Azure/azure-sdk-for-java/issues/3169 tracking that work